neuraxle/metaopt/auto_ml.py

```diff
@@ -279,323 +419,557 @@ def find_next_best_hyperparams(self, auto_ml_container: 'AutoMLContainer') -> Hy
+
+
+class AutoML(ForceHandleOnlyMixin, BaseStep):
     """
     A step to execute any Automatic Machine Learning Algorithms.
 
     Example usage :
 
     .. code-block:: python
 
-        auto_ml: AutoMLSequentialWrapper = AutoMLSequentialWrapper(
-            step=ForecastingPipeline(),
-            auto_ml_algorithm=AutoMLAlgorithm(
-                hyperparameter_optimizer=RandomSearchHyperparameterOptimizer(),
-                validation_technique=KFoldCrossValidationWrapper(),
-                higher_score_is_better=True
+        auto_ml = AutoML(
+            pipeline=Pipeline([
+                MultiplyByN(2),
+                NumpyReshape(shape=(-1, 1)),
+                linear_model.LinearRegression()
+            ]),
+            validation_technique=KFoldCrossValidationWrapper(
+                k_fold=2,
+                scoring_function=average_kfold_scores(mean_squared_error),
+                split_data_container_during_fit=False,
+                predict_after_fit=False
             ),
-            hyperparams_repository=HyperparamsJSONRepository(),
-            n_iters=100
+            hyperparams_optimizer=RandomSearchHyperparameterOptimizer(),
+            hyperparams_repository=InMemoryHyperparamsRepository(),
+            scoring_function=average_kfold_scores(mean_squared_error),
+            n_trial=1,
+            metrics={'mse': average_kfold_scores(mean_squared_error)},
+            epochs=2
         )
-        auto_ml: AutoMLSequentialWrapper = auto_ml.fit(data_inputs, expected_outputs)
-        best_model: ForecastingPipeline = auto_ml.get_best_model()
 
+        auto_ml = auto_ml.fit(data_inputs, expected_outputs)
 
     .. seealso::
-        :class:`AutoMLAlgorithm`,
-        :class:`HyperparamsRepository`,
-        :class:`RandomSearch`,
+        :class:`BaseValidation`,
         :class:`BaseHyperparameterOptimizer`,
+        :class:`HyperparamsRepository`,
         :class:`RandomSearchHyperparameterOptimizer`,
-        :class:`MetaStepMixin`,
-        :class:`NonTransformableMixin`,
+        :class:`ForceHandleOnlyMixin`,
         :class:`BaseStep`
     """
 
     def __init__(
             self,
-            wrapped: BaseStep,
-            auto_ml_algorithm: AutoMLAlgorithm,
+            pipeline,
+            validation_technique: BaseCrossValidationWrapper,
+            refit_trial,
+            hyperparams_optimizer: BaseHyperparameterOptimizer = None,
             hyperparams_repository: HyperparamsRepository = None,
-            n_iters: int = 100,
-            refit=True,
+            scoring_function: Callable = None,
+            n_trials: int = 10,
+            metrics: Dict = None,
+            epochs: int = 10,
+            callbacks: List[BaseCallback] = None,
+            refit_callbacks: List[BaseRefitCallback] = None,
+            refit_scoring_function: Callable = None,
+            higher_score_is_better=False,
+            print_func: Callable = None,
```

Comment:
Might change to a logger class later one. Must resolve to issue.

Reply:
https://github.com/Neuraxio/Neuraxle/issues/283